pkg/rtc/subscriptionmanager.go

```diff
@@ -293,13 +298,14 @@ func (m *SubscriptionManager) reconcileSubscription(s *trackSubscription) {
 				if s.durationSinceStart() > subscriptionTimeout {
 					s.maybeRecordError(m.params.Telemetry, m.params.Participant.ID(), err, true)
 				}
-			case ErrTrackNotFound:
+			case ErrTrackNotFound, ErrSubscriptionLimitExceeded:
```

Comment:
we should *not* change intent of the user in this condition. it's important for desired state to match the client.

instead we should avoid attempting subscription of that type if it's over limit.